src/fonduer/features/feature_libs/tabular_features.py

```diff
@@ -133,53 +130,52 @@ def _tablelib_unary_features(span: SpanMention) -> Iterator[Tuple[str, int]]:
         #      yield "COL_INFERRED_%s_[%s]" % (attrib.upper(), ngram), DEF_VALUE
 
 
-def _tablelib_binary_features(
-    span1: SpanMention, span2: SpanMention
+def _tablelib_multary_features(
+    spans: Tuple[SpanMention, ...]
 ) -> Iterator[Tuple[str, int]]:
-    """Table-/structure-related features for a pair of spans."""
+    """Table-/structure-related features for multiple spans."""
     binary_features = settings["featurization"]["tabular"]["binary_features"]
-    if span1.sentence.is_tabular() and span2.sentence.is_tabular():
-        if span1.sentence.table == span2.sentence.table:
+    span_sentences = [span.sentence for span in spans]
+    if all([sentence.is_tabular() for sentence in span_sentences]):
+        span_tables = [sentence.table for sentence in span_sentences]
+        if [span_tables[1:] == span_tables[:-1]]:
             yield "SAME_TABLE", DEF_VALUE
-            if span1.sentence.cell is not None and span2.sentence.cell is not None:
+            if all([span.sentence.cell is not None for span in spans]):
                 row_diff = min_row_diff(
-                    span1.sentence,
-                    span2.sentence,
+                    span_sentences,
                     absolute=binary_features["min_row_diff"]["absolute"],
                 )
                 col_diff = min_col_diff(
-                    span1.sentence,
-                    span2.sentence,
+                    span_sentences,
                     absolute=binary_features["min_col_diff"]["absolute"],
                 )
                 yield f"SAME_TABLE_ROW_DIFF_[{row_diff}]", DEF_VALUE
                 yield f"SAME_TABLE_COL_DIFF_[{col_diff}]", DEF_VALUE
                 yield (
                     f"SAME_TABLE_MANHATTAN_DIST_[{abs(row_diff) + abs(col_diff)}]"
                 ), DEF_VALUE
-                if span1.sentence.cell == span2.sentence.cell:
+                span_cells = [sentence.cell for sentence in span_sentences]
+                if [span_cells[1:] == span_cells[:-1]]:
                     yield "SAME_CELL", DEF_VALUE
-                    yield (
-                        f"WORD_DIFF_["
-                        f"{span1.get_word_start_index() - span2.get_word_start_index()}"
-                        f"]"
-                    ), DEF_VALUE
-                    yield (
-                        f"CHAR_DIFF_[{span1.char_start - span2.char_start}]"
-                    ), DEF_VALUE
-                    if span1.sentence == span2.sentence:
+                    word_diff = spans[0].get_word_start_index() - min(
```

Comment:
Why spans[0].get_word_start_index() - min(span.get_word_start_index() for span in spans)? This is not equivalent to the binary case.

Reply:
Yeah that was totally wrong, I did some update there, could you please re-check to see if it follows the expected logic or not ?